src/core/core.controller.js

```diff
@@ -457,7 +457,7 @@ module.exports = function(Chart) {
 
 			// Draw each dataset via its respective controller (reversed to support proper line stacking)
 			helpers.each(me.data.datasets, function(dataset, datasetIndex) {
-				if (me.isDatasetVisible(datasetIndex)) {
+				if (me.isDatasetVisible(datasetIndex) && me.isDatasetVisible(datasetIndex).hasOwnProperty('controller')) {
```

Comment:
The data *implicit* synchronization only alters the meta-data (`getDatasetMeta(datasetIndex)().data` array), not the dataset meta object, so `controller` should not be null here. Anyway, a proper fix need to be done on the issue origin, else we will have this workaround at many places in the code (need a test case).